aion-docker.py

```diff
@@ -0,0 +1,99 @@
+            for peer in peers:
+                add_child_text(element, 'node', 'node', peer)
+
+# pretty printing does not work with ElementTree
+# use this function after inserting new elements in the xml file
+def indent(elem, level=0):
+    i = "\n" + level*indentation
+    if len(elem):
+        if not elem.text or not elem.text.strip():
+            elem.text = i + indentation
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+        for elem in elem:
+            indent(elem, level+1)
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+    else:
+        if level and (not elem.tail or not elem.tail.strip()):
+            elem.tail = i
+
+
+et = xml.etree.ElementTree.parse(config_file)
+root = et.getroot()
+
+api = root.find('api')
+new_rpc = api.find('rpc')
+override_attrib(new_rpc, 'ip', 'rpc_listen_address', os.environ.get('rpc_listen_address'))
+override_attrib(new_rpc, 'port', 'rpc_listen_port', os.environ.get('rpc_listen_port'))
+override_child_text(new_rpc, 'cors-enabled', 'cors_enabled', os.environ.get('cors_enabled'))
+override_child_text(new_rpc, 'apis-enabled', 'apis_enabled', os.environ.get('apis_enabled'))
+
+new_java = api.find('java')
+override_attrib(new_java, 'ip', 'java_api_listen_address', os.environ.get('java_api_listen_address'))
+override_attrib(new_java, 'port', 'java_api_listen_port', os.environ.get('java_api_listen_port'))
+
+net = root.find('net')
+new_p2p = net.find('p2p')
+override_child_text(new_p2p, 'ip', 'p2p_listen_address', os.environ.get('p2p_listen_address'))
+override_child_text(new_p2p, 'port', 'p2p_listen_port', os.environ.get('p2p_listen_port'))
+override_child_text(new_p2p, 'discover', 'discover', os.environ.get('discover'))
+
+new_nodes = net.find('nodes')
+add_peers(new_nodes, os.environ.get('override_peer_list'), os.environ.get('peer_list'))
+
+new_consensus = root.find('consensus')
+override_child_text(new_consensus, 'mining', 'mining', os.environ.get('mining'))
+override_child_text(new_consensus, 'miner-address', 'miner_address', os.environ.get('miner_address'))
+
+new_log = root.find('log')
+override_child_text(new_log, 'log-file', 'log_file', os.environ.get('log_file'))
+override_child_text(new_log, 'log-path', 'log_path', os.environ.get('log_path'))
+override_child_text(new_log, 'GEN', 'log_level_gen', os.environ.get('log_level_gen'))
+override_child_text(new_log, 'VM', 'log_level_vm', os.environ.get('log_level_vm'))
+override_child_text(new_log, 'API', 'log_level_api', os.environ.get('log_level_api'))
+override_child_text(new_log, 'SYNC', 'log_level_sync', os.environ.get('log_level_sync'))
+override_child_text(new_log, 'CONS', 'log_level_cons', os.environ.get('log_level_cons'))
+override_child_text(new_log, 'DB', 'log_level_db', os.environ.get('log_level_db'))
+
+indent(root)
+et.write(config_file, encoding='utf-8', xml_declaration=True)
+
+subprocess.call(['/bin/bash', '-c', './aion.sh'])
```

Comment:
should allow user to set parameters when executing the kernel; ./aion.sh [options]